### PR TITLE
[PREVIEW] RDM 2773 Write to Azure Blob Store - Happy Path

### PR DESCRIPTION
**Pull request by adr1ancho**
### JIRA link (if applicable) ###
RDM-2773

### Change description ###
Write to Azure BS

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: application/src/main/java/uk/gov/hmcts/dm/service/BlobStorageWriteService.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                                             @NotNull MultipartFile multiPartFile) {
+        writeBinaryStream(storedDocument.getId(), documentContentVersion, multiPartFile);
+        documentContentVersionRepository.update(documentContentVersion.getId(), documentContentVersion.getContentUri());
+    }
+
+    private void writeBinaryStream(UUID documentId,
+                                   DocumentContentVersion documentContentVersion,
+                                   MultipartFile multiPartFile) {
+        try {
+            CloudBlockBlob blob = getCloudFile(defaultIfNull(documentContentVersion.getId(), UUID.randomUUID());
```
Comment (rafalkalita) [inline application/src/main/java/uk/gov/hmcts/dm/service/BlobStorageWriteService.java]: Missing ')' at the end.

File: application/src/main/java/uk/gov/hmcts/dm/service/BlobStorageMigrationService.java [hunk 2]
```diff
@@ -67,15 +67,17 @@ private DocumentContentVersion getDocumentContentVersion(final @NotNull UUID doc
...
     private void uploadBinaryStream(UUID documentId, DocumentContentVersion dcv) {
         try {
-            CloudBlockBlob blob = getCloudFile(dcv.getId());
-            blob.upload(dcv.getDocumentContent().getData().getBinaryStream(), dcv.getSize());
-            dcv.setContentUri(blob.getUri().toString());
+            CloudBlockBlob cloudBlockBlob = getCloudFile(dcv.getId());
+            Blob data = dcv.getDocumentContent().getData();
+            cloudBlockBlob.upload(data.getBinaryStream(), dcv.getSize());
+            dcv.setContentUri(cloudBlockBlob.getUri().toString());
+            dcv.setContentChecksum(Sha512DigestUtils.shaHex(data.getBytes(1, (int)data.length())));
```
Comment (rafalkalita) [inline application/src/main/java/uk/gov/hmcts/dm/service/BlobStorageMigrationService.java]: Not unit tested. Remove the line and run the test. All is green.

File: application/src/main/java/uk/gov/hmcts/dm/service/BlobStorageDeleteService.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.dm.domain.DocumentContentVersion;
+import uk.gov.hmcts.dm.exception.FileStorageException;
+
+import java.net.URISyntaxException;
+import java.util.UUID;
+
+@Service
+public class BlobStorageDeleteService {
```
Comment (rafalkalita) [inline application/src/main/java/uk/gov/hmcts/dm/service/BlobStorageDeleteService.java]: Where is the unit test?

**Comment by adr1ancho** [comment]
Stale PR

**Pull request closed without merging**